src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -678,4 +678,25 @@ object messages {
            |$fixedVarInAlternative""".stripMargin
     }
   }
+
+  case class AuxConstructorNeedsNonImplicitParameter()(implicit ctx:Context) extends Message(24) {
+    val kind = "Syntax"
+    val msg = "auxiliary constructor needs non-implicit parameter list"
+    val code =
+      s"""class Person(val first: String, val last: String) {
+          |  def this() = this("John", "Doe")
+          |  def this(last: String) = this("John", last)
+          |}""".stripMargin
+
+    val explanation =
+      hl"""An auxiliary constructor requires a non-implicit parameter list. Auxiliary
+        |constructors are allowed to call either auxiliary or primary constructors,
+        |where the primary constructor is allowed implicit parameters. A common mistake
+        |is to forget the parenthesis on `this` (eg `def this() = ...`).
+        |
+        |Here's an example class with primary and auxiliary constructors:
+        |
+        |$code""".stripMargin
```

Comment:
I'm a bit over my head (learning scala from coursera, on week 7) with creating an example for the other case. I've been looking at `math.Ordering` and I cannot create a simple example; and my example in this PR actually does against the general consensus of not using basic types in `implicit`. Do you think I should remove the example and not have one? I'm considering adding more text to the explanation to help with the case of an implicit primary constructor, thoughts @felixmulder?